spicy/toolchain/src/compiler/validator.cc

```diff
@@ -313,6 +313,11 @@ struct VisitorPost : visitor::PreOrder, hilti::validator::VisitorMixIn {
                     error("declaration needs a concrete struct type", n, node::ErrorPriority::High);
             }
         }
+
+        if ( auto unit_ty = n->type()->type()->template tryAs<type::Unit>() ) {
+            if ( ! unit_ty->typeID() )
+                error("unit types must be named", n->type());
```

Comment:
why is this check (in `checkVariable`) necessary, even though there is a check in the `operator()(spicy::type::Unit*)` method for this too? Is it not resolved at this point? The `ast-orig` does not show the unit type for the local/global fields either

Reply:
Actually, I'm not sure either. I would also expect this to be unnecessary. And yes, looking at the AST output, that does look weird, not sure right now what's going on there.

Reply:
Ok, I figured this one out. It's hard to see in the AST output but the qtype is an *external* qtype, meaning that rather having a type as its child, it's just pointing to type that already exists elsewhere. The assumption is that that existing type is part of the AST as well and validated there at its original location. However, turns out that assumption is not right here: the fact that it's an external type is just an artifact of how the parser creates the qtype, with the type we're pointing to not ending up in the AST itself. 

I'll add a commit at the beginning of the branch that fixes this; it's unnecessary & inefficient anyways. And I'll tie remove this block.